test/common/buffer/zero_copy_input_stream_test.cc

```diff
@@ -0,0 +1,79 @@
+#include "common/buffer/buffer_impl.h"
+#include "common/buffer/zero_copy_input_stream_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Buffer {
+namespace {
+
+class ZeroCopyInputStreamTest : public testing::Test {
+public:
+  ZeroCopyInputStreamTest() {
+    Buffer::OwnedImpl buffer{"abcd"};
+    stream_.move(buffer);
+  }
+
+  std::string slice_data_{"abcd"};
+  ZeroCopyInputStreamImpl stream_;
+
+  const void* data_;
+  int size_;
```

Comment:
`data_` and `size_` are used to receive output from `Next`, so types here matches the interface.